dss/events/handlers/index.py

```diff
@@ -6,43 +6,58 @@
 import uuid
 from urllib.parse import unquote
 
-import boto3
-import botocore
 import requests
 from elasticsearch.helpers import scan
 
+from dss import Config
 from ... import (DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE,
                  DSS_ELASTICSEARCH_QUERY_TYPE, DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
                  DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE)
 from ...util import create_blob_key
 from ...hcablobstore import BundleMetadata, BundleFileMetadata
 from ...util.es import ElasticsearchClient
+from ...blobstore import BlobStore, BlobNotFoundError
 
 DSS_BUNDLE_KEY_REGEX = r"^bundles/[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}\..+$"
 
 
-def process_new_indexable_object(event, logger) -> None:
+def process_new_s3_indexable_object(event, logger) -> None:
     try:
         # This function is only called for S3 creation events
         key = unquote(event['Records'][0]["s3"]["object"]["key"])
-        if is_bundle_to_index(key):
-            logger.info(f"Received S3 creation event for bundle which will be indexed: {key}")
-            s3 = boto3.resource('s3')
-            bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
-            manifest = read_bundle_manifest(s3, bucket_name, key, logger)
-            bundle_id = get_bundle_id_from_key(key)
-            index_data = create_index_data(s3, bucket_name, bundle_id, manifest, logger)
-            add_index_data_to_elasticsearch(bundle_id, index_data, logger)
-            subscriptions = find_matching_subscriptions(index_data, logger)
-            process_notifications(bundle_id, subscriptions, logger)
-            logger.debug(f"Finished index processing of S3 creation event for bundle: {key}")
-        else:
-            logger.debug(f"Not indexing S3 creation event for key: {key}")
+        bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
+        process_new_indexable_object(bucket_name, key, "aws", logger)
     except Exception as ex:
         logger.error(f"Exception occurred while processing S3 event: {ex} Event: {json.dumps(event, indent=4)}")
         raise
 
 
+def process_new_gs_indexable_object(event, logger) -> None:
```

Comment:
Did I miss the place where this is called?

Reply:
I'll remove for now and add it in my next PR, since test coverage is not there. Otherwise it won't pass the code coverage.